app/src/main/java/org/mozilla/fenix/gleanplumb/DefaultMessageController.kt

```diff
@@ -48,6 +50,7 @@ class DefaultMessageController(
         if (message.maxDisplayCount <= message.metadata.displayCount + 1) {
             Messaging.messageExpired.record(Messaging.MessageExpiredExtra(message.id))
         }
+        messagingFeature.recordExposure()
```

Comment:
If there are three messages in the system: 

- message `FML` is hardcoded into the `nimbus.fml.yaml` file.
- message `EXP` is from an experiment
- message `ROL` is from a rollout.

If a `FML` or `ROL` are displayed, the message system  will `recordExposure` here, erroneously.

(If `EXP` doesn't exist, i.e. there are no messages under experiment, this is ok, but when there are, you over record exposure to the treatment even when you're not displaying the treatment).

Reply:
I think you're also recording exposure twice in the case of `EXP`, since you record exposure when the message is selected.

Recording at message selection time is good, because it's a good time to deal with control message.

I guess one good way of doing this would be to add a `recordExposure` boolean to the Message state itself, so that you only call into Nimbus when the message is displayed, but the decision whether to record or not is made at message selection time.

Reply:
My mistake, it looks like I didn't committed the change, where I removing the `recordExposure` from where we are selecting the message.

Reply:
Could help me to clarify what is a "message ROL is from a rollout"?

Reply:
> message EXP is from an experiment
Do an message EXP should be shown to the user?